From c3cfbb3930c6f835f7cafb0c24889801ad1640c3 Mon Sep 17 00:00:00 2001
From: kwan1996 <laicheehou9@gmail.com>
Date: Thu, 31 Jul 2025 11:48:15 +0800
Subject: [PATCH 4/7] Fix timer.h build error

---
 backport-include/linux/timer.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backport-include/linux/timer.h b/backport-include/linux/timer.h
index 1efa673..20c4467 100644
--- a/backport-include/linux/timer.h
+++ b/backport-include/linux/timer.h
@@ -79,7 +79,7 @@ static inline int timer_delete_sync(struct timer_list *timer)
 }
 #endif /* < 6.1.84 */
 
-#if LINUX_VERSION_IS_LESS(6,2,0)
+#if LINUX_VERSION_IS_LESS(5,4,0) // < original < (6,2,0)
 static inline int timer_shutdown_sync(struct timer_list *timer)
 {
 	return del_timer_sync(timer);
-- 
2.34.1

