From be798eb3348f24f6f5f850703d740a59b3b39bd4 Mon Sep 17 00:00:00 2001
From: kwan1996 <laicheehou9@gmail.com>
Date: Thu, 31 Jul 2025 11:50:33 +0800
Subject: [PATCH 5/7] Fix ath12k driver build error for 5.15

---
 drivers/net/wireless/ath/ath12k/pci.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/pci.c b/drivers/net/wireless/ath/ath12k/pci.c
index 14954bc..876f465 100644
--- a/drivers/net/wireless/ath/ath12k/pci.c
+++ b/drivers/net/wireless/ath/ath12k/pci.c
@@ -784,7 +784,11 @@ static int ath12k_pci_msi_alloc(struct ath12k_pci *ab_pci)
 	}
 
 	ab_pci->msi_ep_base_data = msi_desc->msg.data;
+#if LINUX_VERSION_IS_GEQ(5,17,0)
 	if (msi_desc->pci.msi_attrib.is_64)
+#else
+	if (msi_desc->msi_attrib.is_64)
+#endif
 		set_bit(ATH12K_PCI_FLAG_IS_MSI_64, &ab_pci->flags);
 
 	ath12k_dbg(ab, ATH12K_DBG_PCI, "msi base data is %d\n", ab_pci->msi_ep_base_data);
-- 
2.34.1

