From 7bc3871a56b1349fa538aa5348e0bb094099d11b Mon Sep 17 00:00:00 2001
From: kwan1996 <laicheehou9@gmail.com>
Date: Thu, 31 Jul 2025 11:56:25 +0800
Subject: [PATCH 7/7] Fix mhi build error for 5.15

---
 drivers/bus/mhi/host/init.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/bus/mhi/host/init.c b/drivers/bus/mhi/host/init.c
index 44f9349..06d8b20 100644
--- a/drivers/bus/mhi/host/init.c
+++ b/drivers/bus/mhi/host/init.c
@@ -1403,7 +1403,7 @@ void mhi_driver_unregister(struct mhi_driver *mhi_drv)
 }
 EXPORT_SYMBOL_GPL(mhi_driver_unregister);
 
-static int mhi_uevent(const struct device *dev, struct kobj_uevent_env *env)
+static int mhi_uevent(struct device *dev, struct kobj_uevent_env *env)
 {
 	const struct mhi_device *mhi_dev = to_mhi_device(dev);
 
-- 
2.34.1

